### ISSUE TITLE
Add a new column and update it, Data view is not refreshed

### ISSUE THREAD

**Issue by andrewbt** [CartoDB/cartodb#5291]
Steps to reproduce bug:
1. In the SQL pane, issue an `ALTER TABLE tablename ADD COLUMN newcolumn numeric` query. Every row of that new column will be null and will display as such in the Data View.
2. I want to populate this new column with derivative values from another column (text based percentage strings like "93.45%", I want to make them numeric), so I next issue an `UPDATE tablename SET newcolumn = cast(trim(trailing '%' from tablename.textpercentcol) as numeric)` query in the SQL pane.
3. After the query finishes, the Data View still shows every row in my new column as "null". It was only after much wandering around and finally issuing a `SELECT * FROM tablename WHERE newcolumn is not null` where I realized the update had finally actually occurred and these columns were, in fact, populated.

**Comment by andrewbt** [comment]
Only seems to be the first 40 rows. Might be a caching/CDN problem, but I've hard refreshed the page several times and about 20 minutes has gone past since the first query, so if so our caching isn't updating very fast.

![null_column](https://cloud.githubusercontent.com/assets/2013555/9554204/43fe8512-4d92-11e5-9c87-e33ad26180c4.gif)

**Comment by andrewbt** [comment]
Changing the names or datatypes of columns through the UI also runs into the same problem - the changes are not reflected in the UI but have taken place in the database. Refreshing the Data View or waiting a few minutes does not resolve the problem.

**Comment by luisbosque** [comment]
Have you tried to run the SELECT query manually to the SQL API? It would be nice trying to run it with v1 and v2 and see if there is any difference in order to check if it's a caching problem or not

**Comment by andrewbt** [comment]
@luisbosque @xavijam I ran into a similar problem with a new dataset (query "alter table add column" but the new column doesn't appear in the Editor; nor does it with the "add column" button) and tested with the SQL API v1 and v2 - the columns both appear in the API responses. But not the Editor.

**Comment by d3dc** [comment]
Was there any progress on this bug?

We just spent several hours debugging why our named map's interactivity configuration wasn't working, only to notice that our browsers were displaying different columns based on when we first accessed the data view.

Clearing all browser data and relogging seems to have updated the data view. The changes to column names happened 3 days ago.

**Comment by kyaroch** [comment]
I've encountered a problem that may be related. Using the Editor, I renamed a column and created a new column. The changes are visible in SQL queries I run in the Editor, but not visible in the Map View or in the Data View in the absence of a query. I can't edit the renamed column in the GUI, reference it in CartoCSS under either name, or scroll down in the Data View; these all produce error messages.

I was not able to fix this problem by:
* Refreshing the Editor
* Logging out and back in again
* Clearing the browser cache
* Using a different browser
* Using a different computer
* Using a different Internet connection

So far the only solution has been to create a new dataset from a `SELECT * FROM table_name` query. Apologies if I'm not describing this as clearly as I should be.
